### DISCUSSION TITLE
Transition opacity to 50%?

### DISCUSSION THREAD

**Discussion by JaredCostall** [alpinejs/alpine#3598] [7. Bugs (archived)] [answered]
With the following, I would expect the element to fade to 50%, yet it fades to 50% and then jumps to 100% at the end of the specified duration.

Also adding a default class of `opacity-50` does not induce the desired effect.

```
<div x-data="{ open: false }">
        <button @click="open = ! open">Toggle</button>
        <div x-show="open" 
            x-transition:enter="transition ease-out duration-1000"
            x-transition:enter-start="opacity-0"
            x-transition:enter-end="opacity-50"
            x-transition:leave="transition ease-in duration-1000"
            x-transition:leave-start="opacity-50"
            x-transition:leave-end="opacity-0">
                Hello 👋
        </div>
</div>
```

**Comment by ekwoka** [comment, marked as answer]
The transition classes are only there while the transition is happening.

Basically the start ones and the non labeled ones are added. Then one tick passes, the start ones are removed and the end ones are added. Once transition ends, they're all removed.

What you should do is put opacity 50 on the element, not in the transition classes.

And maybe give the opacity-0 in the transitions a `!` prefix

**Reply by josefrichter** [reply]
The need to add `!` prefix, as a result of the explanation how it works, is such a crucial piece of information, that it ought to be in the docs.

**Reply by ekwoka** [reply]
Tough, since that's about how Tailwind works more than it is about how Alpine works...
